File: LeetCode/Simulation/Python/Leetcode735-asteroid-collision.py
class Solution:
    def asteroidCollision(self, asteroids: List[int]) -> List[int]:
        ans = []
        for ast in asteroids:
            flag = True
            while flag and ans and ans[-1] > 0 and ast < 0:
                flag = -ast > ans[-1]
                if -ast >= ans[-1]: ans.pop()
            if flag: ans.append(ast)
        return ans

File: LeetCode/Simulation/Rust/Leetcode735-asteroid-collision.rs
impl Solution {
  pub fn asteroid_collision(asteroids: Vec<i32>) -> Vec<i32> {
    let mut ans = vec![];
    for &ast in &asteroids {
      let mut flag = true;
      while flag && ast < 0 && !ans.is_empty() && ans[ans.len() - 1] > 0 {
        flag = -ast > ans[ans.len() - 1];
        if -ast >= ans[ans.len() - 1] {
          ans.pop();
        }
      }
      if flag {
        ans.push(ast);
      }
    }
    ans
  }
}
